ament_cpplint/ament_cpplint/main.py

```diff
@@ -144,7 +148,8 @@ def main(argv=sys.argv[1:]):
         else:
             print("Not using '--root'")
         print('')
-        arguments += files
+        arguments += [file for file in files
+                      if any(excl not in file or not excl for excl in args.exclude)]
```

Comment:
Can you explain the logic of this condition? `excl not in file` looks wrong to me.

Reply:
`if excl is not in the file` -> this means that if the `excl` file or path is not in the the "include" filenames or paths then should **not** be discarded

Reply:
So `file` (which is btw a `builtin` and therefore shouldn't be used as a variable name) is the path of a to-be-checked file, right? E.g. `foo/bar/baz.ext`.

Why should an `excl` value `bar` skip that file? When passing `--exclude x` that would exclude every file which contains an `x`?

Reply:
> Why should an excl value bar skip that file? 

I was trying to simplify it. Instead of writing the whole path, just writing the folder o the filename

> When passing --exclude x that would exclude every file which contains an x?
yes, you are right